### PR TITLE
Fix type dependencies of mypy invocation

### PR DESCRIPTION
**Pull request by hukkin**
Removes `types-toml` from pre-commit mypy invocation's addiditional dependencies. Adds `tomli` and `platformdirs`.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -25,9 +25,10 @@ repos:
         exclude: ^docs/conf.py
         additional_dependencies:
           - types-dataclasses >= 0.1.3
-          - types-toml >= 0.1.1
+          - tomli >= 0.2.6
```
Comment (ichard26) [inline .pre-commit-config.yaml:28]: Do you want an upper bound here too?

File: Pipfile [hunk 2]
```diff
@@ -11,7 +11,6 @@ flake8 = "*"
 flake8-bugbear = "*"
 mypy = ">=0.812"
 types-dataclasses = ">=0.1.3"
-types-toml = ">=0.1.1"
```
Comment (ichard26) [inline Pipfile:14]: Good catch!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (hukkin) [reply .pre-commit-config.yaml]: :+1: changed it for consistency now that you mentioned. Though it's just for mypy so shouldn't matter too much :smile_cat:

**Review by ichard26 — APPROVED** [review]
Thank you!

**Pull request merged**